matter_persistence/redis/manager.py

```diff
@@ -234,25 +249,55 @@ async def get_with_key(self, key: str, object_class: type[Model] | None = None,
         return value
 
     async def get_many_with_keys(
-        self, keys: Sequence[str], object_class: type[Model] | None = None
+            self, keys: Sequence[str], object_class: type[Model] | None = None, use_key_as_is: bool = False
     ) -> dict[str, bytes | list[bytes] | Model | list[Model]]:
-        object_name = object_class.__name__ if object_class else None
         return_set: dict[str, bytes | list[bytes] | Model | list[Model]] = {}
-        async with self.__get_cache_client(for_writing=False) as cache_client:
-            processed_input = {
-                CacheHelper.create_basic_hash_key(original_key, object_name): original_key for original_key in keys
-            }
-            response: dict[str, bytes | list[bytes]] = await cache_client.get_many_values(processed_input.keys())
+
+        if use_key_as_is:
+            async with self.__get_cache_client(for_writing=False) as cache_client:
+                response: dict[str, bytes | list[bytes]] = await cache_client.get_many_values(keys)
+
+            if object_class:
+                for key, value in response.items():
+                    if value:
+                        value_obj = json.loads(value.decode("utf-8"))
+                        if isinstance(value_obj, list):
+                            return_set[key] = [object_class.model_validate(item) for item in
+                                                                value_obj]
+                        elif value_obj is not None:
+                            return_set[key] = object_class.model_validate(value_obj)
+                        else:
+                            return_set[key] = value_obj
+                    else:
+                        # Returns None
+                        return_set[key] = value
+            else:
+                return_set = {key: value for key, value in response.items()}
+        else:
+            object_name = object_class.__name__ if object_class else None
+            async with self.__get_cache_client(for_writing=False) as cache_client:
+                processed_input = {
+                    CacheHelper.create_basic_hash_key(original_key, object_name): original_key for original_key in keys
+                }
+                response: dict[str, bytes | list[bytes]] = await cache_client.get_many_values(processed_input.keys())
+
```

Comment:
Please refactor this to extract creating keys inside the first `if` & leave the rest as-is.

Something like:

```python
if use_key_as_is:
    keys = {key: key for key in keys}
else:
    keys = {CacheHelper.create_basic_hash_key(original_key, object_name): original_key for original_key in keys}

# proceed to use those
```